### ISSUE TITLE
Tracking Issue for `const_make_ascii`

### ISSUE THREAD

**Issue by bjoernager** [rust-lang/rust#130698]
Feature gate: `#![feature(const_make_ascii)]`

This is a tracking issue for marking the `make_ascii_uppercase` and `make_ascii_lowercase` methods in `char`, `u8`, `[u8]`, and `str` as "const."

### Public API

```rust
impl char {
    pub const fn make_ascii_uppercase(&mut self);

    pub const fn make_ascii_lowercase(&mut self);
}

impl u8 {
    pub const fn make_ascii_uppercase(&mut self);

    pub const fn make_ascii_lowercase(&mut self);
}

impl [u8] {
    pub const fn make_ascii_uppercase(&mut self);

    pub const fn make_ascii_lowercase(&mut self);
}

impl str {
    pub const fn make_ascii_uppercase(&mut self);

    pub const fn make_ascii_lowercase(&mut self);
}
```

### Steps / History

- [x] Implementation for `char`: #130697
- [x] Implementation for `u8`: #130713
- [x] Implementation for `[u8]` and `str`: #130738
- [x] Final comment period (FCP)
- [ ] Stabilization PR

### Unresolved Questions

- None yet.

**Comment by dtolnay** [comment]
@rust-lang/libs-api:
@rfcbot fcp merge

`to_ascii_uppercase`/`to_ascii_lowercase` have been stable in const since 1.52. `make_ascii_uppercase`/`make_ascii_lowercase` are unblocked now by the stabilization of https://github.com/rust-lang/rust/issues/57349.

**Comment by okaneco** [comment]
Can the `u8` methods be included in this?
https://doc.rust-lang.org/stable/std/primitive.u8.html#method.make_ascii_lowercase
https://doc.rust-lang.org/stable/std/primitive.u8.html#method.make_ascii_uppercase

Similarly const stable `u8::to_ascii_uppercase`/`u8::to_ascii_lowercase` since 1.52.

**Comment by bjoernager** [comment]
Yes, that sounds like a good idea. I'll try adding another PR later.

**Comment by bjoernager** [comment]
#131496 stabilises this.
